Feature/netcms 58 query pool

sschmittBt opened the pull request:
Added query pool controller to execute queries from a preloaded pool when calling specific pages

sschmittBt committed:
NetCMS 58: Query Pool
-added query pool controller
-added unit tests
-added JSON to POJO classes for saved queries

sschmittBt committed:
Merged 'develop'.

sschmittBt committed:
Increased vertx version
changed query calls to new query implementation

sschmittBt committed:
Merged 'develop'.

# Conflicts:
#	gradle.properties
#	src/main/java/de/braintags/netrelay/MemberUtil.java
#	src/main/java/de/braintags/netrelay/controller/api/DataTableLinkDescriptor.java
#	src/main/java/de/braintags/netrelay/controller/authentication/PasswordLostController.java
#	src/main/java/de/braintags/netrelay/controller/authentication/RegisterController.java
#	src/main/java/de/braintags/netrelay/controller/persistence/RecordContractor.java
#	src/test/java/de/braintags/netrelay/unit/NetRelayBaseConnectorTest.java
#	src/test/java/de/braintags/netrelay/unit/TAuthenticationMongoEncrypted.java
#	src/test/java/de/braintags/netrelay/unit/TPasswordLost.java
#	src/test/java/de/braintags/netrelay/unit/TRegistration.java
#	src/test/java/de/braintags/netrelay/unit/persistence/TPersistenceController_Delete.java

sschmittBt committed:
QueryPool implementation
-changed to new query implementation
-added unit tests for query building
-added unit tests for controller call for dynamic queries
-added rudimentary variable replacement in queries

sschmittBt committed:
Merged 'develop'.

sschmittBt committed:
QueryPool implementation
-added more unit tests for the query pool controller
-added complex variable replacement in queries

sschmittBt committed:
QueryPool Variables and Caching
-included changes from pojo-mapper #54
-changed cache to cache the whole query instead of just the template
-added field condition variable resolver that fetches the values from context, requests or records

sschmittBt committed:
Merged 'develop'.

sschmittBt committed:
Merged 'develop'.

# Conflicts:
#	src/main/java/de/braintags/netrelay/controller/api/DataTableLinkDescriptor.java

sschmittBt committed:
Classpath changes
- fixed references and comments to old pojomapepr classpath

sschmittBt committed:
Pre-Merge cleanup
- Fixed comments and some remnants of older versions
- added tests to test suite

remi128 reviewed (changes requested):
This merge request wants to merge into master, but develop is required

remi128 commented on src/main/java/de/braintags/netrelay/controller/persistence/DisplayAction.java:
```diff
@@ -58,10 +58,22 @@ protected void handleSubobjectEntityDefinition(RoutingContext context, String en
-      Handler<AsyncResult<Void>> handler) {
+  protected void handleQuery(IQuery<?> query, String entityName, RoutingContext context, CaptureMap map,
+      IMapper<?> mapper, Handler<AsyncResult<Void>> handler) {
+    int limit;
```
initial assignment from  query.getDataStore().getDefaultQueryLimit() to limit saves one command and sonar won't complain

sschmittBt replied:
Done

remi128 commented on src/main/java/de/braintags/netrelay/controller/persistence/DisplayAction.java:
```diff
@@ -58,10 +58,22 @@ protected void handleSubobjectEntityDefinition(RoutingContext context, String en
+    if (map.containsKey(PersistenceController.SELECTION_START_CAPTURE_KEY)) {
+      offset = Integer.parseInt(map.get(PersistenceController.SELECTION_START_CAPTURE_KEY));
+    } else
+      offset = 0;
```
see above direct assignment

sschmittBt replied:
Done

remi128 commented on src/main/java/de/braintags/netrelay/controller/querypool/package-info.java:
```diff
@@ -0,0 +1,121 @@
+/**
```
please link it into de.braintags.netrelay.controller.package-info

sschmittBt replied:
Done

remi128 reviewed (changes requested):
tests are running, some minor comments

sschmittBt committed:
Change requests
- moved limit initialization to reduce complexity
- extended package info with current implementation info

sschmittBt committed:
Merged 'develop'.

# Conflicts:
#	src/main/java/de/braintags/netrelay/controller/persistence/DisplayAction.java
